adapters/risc0/src/host.rs

```diff
@@ -69,20 +76,25 @@ fn verify_from_slice<'a>(
     serialized_proof: &'a [u8],
     code_commitment: &Risc0MethodId,
 ) -> Result<&'a [u8], anyhow::Error> {
-    let receipt: Risc0Proof<'a> = bincode::deserialize(serialized_proof)?;
-    verify_with_hal(
-        &risc0_zkp::verify::CpuVerifyHal::<BabyBear, HashSuiteSha256<BabyBear, Impl>, _>::new(
-            &CIRCUIT,
-        ),
-        &code_commitment.0,
-        &receipt.seal,
-        receipt.journal,
-    )?;
-    Ok(receipt.journal)
+    let Risc0Proof::<'a> {
+        segment_receipts,
+        journal,
+        ..
+    } = bincode::deserialize(serialized_proof)?;
+
+    let receipts = segment_receipts
+        .into_iter()
+        .map(|r| r as Box<dyn Receipt>)
+        .collect::<Vec<_>>();
+    SessionReceipt::new(receipts, journal.to_vec()).verify(code_commitment.0)?;
```

Comment:
I think it is ok to do copy of vector now and prioritize upgrade over performance improvement!